xarray/core/dataset.py

```diff
@@ -408,17 +408,9 @@ def __init__(
         self,
         # could make a VariableArgs to use more generally, and refine these
         # categories
-        data_vars: Mapping[
-            Hashable,
-            Union[
-                "DataArray",
-                Variable,
-                Tuple[Hashable, Any],
-                Tuple[Sequence[Hashable], Any],
-            ],
-        ] = None,
+        data_vars: Mapping[Hashable, Any] = None,
         coords: Mapping[Hashable, Any] = None,
-        attrs: Mapping = None,
+        attrs: Mapping[Hashable, Any] = None,
```

Comment:
Support ``Dataset({"x": [1, 2]})``, where one of the data_vars is silently moved to coords